### DISCUSSION TITLE
Imu class and the balancer project

### DISCUSSION THREAD

**Discussion by jeagh** [pybricks/support#923] [Q&A]
I'm wondering if the balancer project by laurensvalk is still compatible with the current version of Pybricks?

Both motors start and continue to run at maximum speed, the rate variable continually varies from around 0 to -4 without any movement of the hub and the angle variable starts at around -1 and increases as time goes by.

I've just begun to experiment with the Imu class.

**Comment by laurensvalk** [comment]
Which hub / design have you tried?

**Reply by laurensvalk** [reply]
Found the issue! For now, you can fix it by mounting the hub upside down.

We'll publish a proper tutorial page eventually.

**Comment by BertLindeman** [comment]
I am not the OP, but .. .. .. 

I just tried balancer from [pybricks-projects](https://github.com/pybricks/pybricks-projects/blob/master/sets/mindstorms-robot-inventor/other-models/balancer/main.py)
with the same symptoms.
This project (no longer?) shows up on [The pybricks projects web pages](https://pybricks.com/projects/)

**Reply by laurensvalk** [reply]
It never did show up there. I'm working on something now, with updated code :smile:

**Comment by BertLindeman** [comment]
Wow, coincidence does not exist 😀 

Thank you, Laurens

**Reply by laurensvalk** [reply]
Telling a software developer that their code is broken is the fastest way to get them to fix things :robot: (and I mean this in a good way. Please keep doing it :smile:)

**Reply by BertLindeman** [reply]
As an old QA-person I know it's only used the wrong way. I used to try all possibilities the wrong way, all the time to proove the software was well built.
👍

**Reply by laurensvalk** [reply]
Nah. More likely the engineer flipped a minus sign.

In fact that's exactly what we did :smile: Good thing this program wasn't officially listed yet. I'll suggest a fix to OP above and later on publish a proper page with updated code.

**Reply by jeagh** [reply]
Thank you Laurens and Bert, I look forward to the revised code. Indecently, I did try starting the hub upside-down but the result was the same. I'll experiment with the code.

**Reply by laurensvalk** [reply]
I didn’t mean starting it upside down, but mounting it upside down in your robot.

**Reply by jeagh** [reply]
What I actually did was hold the robot upside down while I started the programme. I did also try all orientations just in case. I think that's correct, I'll try it again.

More related to my project that led me to your project, are there any plans in the pipeline to implement the IMU heading method?

**Reply by laurensvalk** [reply]
I'm trying to say that won't help. You have to mount the hub upside down in the robot.

That said, fixed code project coming soon.

**Comment by michaelwareman** [comment]
Hello:
I am playing around with the Gelo robot and would like to add the ability for Gelo to sense gestures like a tap/double tap. Is there some basic code that I can play with until better IMU implementations occur.

Mike,

**Reply by dlech** [reply]
There is some basic code to detect movement using the accelerometer in this comment: https://github.com/pybricks/support/issues/933#issue-1567581537
